nautobot/core/tests/integration/test_home.py

```diff
@@ -1,17 +1,145 @@
+        self.load_page(self.live_server_url)
 
-        self.login("bob", "bob")
 
-        # Wait for the page to render and make sure we got a body.
-        self.selenium.wait_for_html("body")
+    def test_homepage_render(self):
+        """
+        Render homepage with app defined objects.
+        """
+        # Set test user to admin
+        self.user.is_superuser = True
+        self.user.save()
+
+        self.load_page(self.live_server_url)
+
+        columns_html = self.selenium.find_element_by_class_name("col-sm-6 col-md-4")
+        for column_idx, column in enumerate(self.layout):
+            for panel_name, panel_details in column.items():
+                columns_html[column_idx].find_element_by_xpath(f".//strong[contains(text(), '{panel_name}')]")
+                for item_name, _ in panel_details.items():
+                    columns_html[column_idx].find_element_by_xpath(f".//a[contains(text(), '{item_name}')]")
+
+
+    def test_homepage_render_counters(self):
+        """
+        Ensure object counters are correct.
+        """
+        # Set test user to admin
+        self.user.is_superuser = True
+        self.user.save()
+
+        self.load_page(self.live_server_url)
+
+        columns_html = self.selenium.find_element_by_class_name("col-sm-6 col-md-4")
+        for column_idx, column in enumerate(self.layout):
+            for panel_name, panel_details in column.items():
+                columns_html[column_idx].find_element_by_xpath(f".//strong[contains(text(), '{panel_name}')]")
+                for item_name, item_details in panel_details.items():
+                    item_html = columns_html[column_idx].find_element_by_xpath(f".//a[contains(text(), '{item_name}')]")
+                    if item_details.get("model"):
+                        counter = item_details["model"].objects.count()
+                        counter_html = int(item_html.find_element_by_xpath(f"./../../span").get_property("innerHTML"))
+                        self.assertEqual(counter, counter_html)
+
+
+    @override_settings(HIDE_RESTRICTED_UI=False)
+    def test_homepage_render_no_permissions(self):
+        """
+        Render homepage with no permissions.
+        """
+        self.load_page(self.live_server_url)
+
+        columns_html = self.selenium.find_element_by_class_name("col-sm-6 col-md-4")
+        for column_idx, column in enumerate(self.layout):
+            for panel_name, panel_details in column.items():
+                columns_html[column_idx].find_element_by_xpath(f".//*[contains(text(), '{panel_name}')]")
+                for item_name, _ in panel_details.items():
+                    item_html = columns_html[column_idx].find_element_by_xpath(f".//h4[contains(text(), '{item_name}')]")
+                    self.assertTrue("mdi mdi-lock" in item_html.find_element_by_xpath(f"./../span").get_property("innerHTML"))
+
+
+    @override_settings(HIDE_RESTRICTED_UI=False)
```

Comment:
Should we also have a test case with `HIDE_RESTRICTED_UI=True` to verify that only the permitted items are rendered and all others are not present?